Interfaces/API/Job.py

```diff
@@ -1057,184 +1058,206 @@ def _handleParameterSequences( self, paramsDict, arguments ):
       paramsDict['Parameters.%s' % pName]['value'] = self.parameterSeqs[pName]
       paramsDict['Parameters.%s' % pName]['type'] = 'JDL'
       if pName in self.wfArguments:
-        arguments.append( ' -p %s=%%(%s)s' % ( self.wfArguments[pName],
-                                               pName ) )
+        arguments.append(' -p %s=%%(%s)s' % (self.wfArguments[pName],
+                                             pName))
 
     return paramsDict, arguments
 
   #############################################################################
-  def _toJDL( self, xmlFile = '', jobDescriptionObject = None ):  # messy but need to account for xml file being in /tmp/guid dir
-    """Creates a JDL representation of itself as a Job.
+  def _toJDL(self, xmlFile='', jobDescriptionObject=None):
+    """ Creates a JDL representation of itself as a Job.
+
+       Example usage:
+
+       >>> job = Job()
+       >>> job._toJDL()
+
+       :param xmlFile: location of the XML file
+       :type xmlFile: str
+       :param jobDescriptionObject: if not None, it must be a StringIO object
+       :type jobDescriptionObject: StringIO
+
+       :returns: JDL (str)
     """
     # Check if we have to do old bootstrap...
-    classadJob = ClassAd( '[]' )
+    classadJob = ClassAd('[]')
 
     paramsDict = {}
     params = self.workflow.parameters  # ParameterCollection object
 
     paramList = params
     for param in paramList:
-      paramsDict[param.getName()] = {'type':param.getType(), 'value':param.getValue()}
+      paramsDict[param.getName()] = {'type': param.getType(), 'value': param.getValue()}
 
     arguments = []
     scriptName = 'jobDescription.xml'
 
     if jobDescriptionObject is None:
       # if we are here it's because there's a real file, on disk, that is named 'jobDescription.xml'
+      # Messy but need to account for xml file being in /tmp/guid dir
       if self.script:
-        if os.path.exists( self.script ):
-          scriptName = os.path.abspath( self.script )
-          self.log.verbose( 'Found script name %s' % scriptName )
+        if os.path.exists(self.script):
+          scriptName = os.path.abspath(self.script)
+          self.log.verbose('Found script name %s' % scriptName)
         else:
-          self.log.error( "File not found", self.script )
+          self.log.warn("File not found", self.script)
       else:
         if xmlFile:
-          self.log.verbose( 'Found XML File %s' % xmlFile )
-          scriptName = xmlFile
-      self.addToInputSandbox.append( scriptName )
+          if os.path.exists(self.script):
```

Comment:
Shouldn't this check for `xmlFile` instead of self.script?

Reply:
yes... thanks!